src/streaming/zarr.stream.cpp

```diff
@@ -0,0 +1,494 @@
+#include "macros.hh"
+#include "zarr.stream.hh"
+#include "acquire.zarr.h"
+
+#include <blosc.h>
+
+#include <filesystem>
+
+namespace fs = std::filesystem;
+
+namespace {
+bool
+is_s3_acquisition(const struct ZarrStreamSettings_s* settings)
+{
+    return nullptr != settings->s3_settings;
```

Comment:
I already mentioned this over Slack. Yoda conditions are uncommon in modern C++ codebases because the compiler and static analyzer will warn you about this. Personally, I find this form less readable and counterintuitive, as it contradicts the natural reading order.